merkletree/pad.go

```diff
@@ -76,39 +77,43 @@ func (pad *PAD) Update(policies Policies) {
 		pad.loadedEpochs = append(pad.loadedEpochs[:0], pad.loadedEpochs[n:]...)
 	}
 
-	pad.updateInternal(policies, pad.currentSTR.epoch+1)
+	pad.updateInternal(policies, pad.latestSTR.epoch+1)
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	cStr := pad.latestSTR
+	index, _ := cStr.computePrivateIndex(key)
+	return pad.tree.Set(index, key, value)
 }
 
-func (pad *PAD) Lookup(key string) *AuthenticationPath {
-	str := pad.currentSTR
-	return str.tree.Get(key)
+func (pad *PAD) Lookup(key string) (*AuthenticationPath, error) {
+	return pad.LookupInEpoch(key, pad.latestSTR.epoch)
 }
 
 func (pad *PAD) LookupInEpoch(key string, epoch uint64) (*AuthenticationPath, error) {
 	str := pad.GetSTR(epoch)
 	if str == nil {
 		return nil, ErrorSTRNotFound
 	}
-	ap := str.tree.Get(key)
+	lookupIndex, proof := str.computePrivateIndex(key)
+	ap := str.tree.Get(lookupIndex)
+	ap.vrfProof = proof
 	return ap, nil
 }
 
 func (pad *PAD) GetSTR(epoch uint64) *SignedTreeRoot {
-	if epoch >= pad.currentSTR.epoch {
-		return pad.currentSTR
+	if epoch >= pad.latestSTR.epoch {
+		return pad.latestSTR
 	}
 	return pad.snapshots[epoch]
 }
 
 func (pad *PAD) TB(key string, value []byte) (*TemporaryBinding, error) {
 	//FIXME: compute private index twice
 	//it would be refactored after merging VRF integration branch
-	index := computePrivateIndex(key)
-	tb := pad.currentSTR.sig
+	cStr := pad.latestSTR
+	index, _ := cStr.computePrivateIndex(key)
```

Comment:
Sure, but I'll only be able to do it later or maybe even tomorrow morning.

Am 22.07.2016 6:55 nachm. schrieb "Vu Quoc Huy" <notifications@github.com>:

> In merkletree/pad.go
> <https://github.com/coniks-sys/coniks-go/pull/19#discussion_r71910352>:
>
> >  	}
> >  	return pad.snapshots[epoch]
> >  }
> >
> >  func (pad *PAD) TB(key string, value []byte) (*TemporaryBinding, error) {
> >  	//FIXME: compute private index twice
> >  	//it would be refactored after merging VRF integration branch
> > -	index := computePrivateIndex(key)
> > -	tb := pad.currentSTR.sig
> > +	cStr := pad.latestSTR
> > +	index, _ := cStr.computePrivateIndex(key)
>
> Can you help me change something here?
> pad.Set(key, value) below to pad.tree.Set(index, key, value) and remove
> the comment above. Many thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coniks-sys/coniks-go/pull/19/files/e5beb0f39c47ff7b153c2cce1507b62551587ac3#r71910352>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHGNZiHzX5-V9B0xLHMUVtNqOtQDehW2ks5qYPYNgaJpZM4JNbH9>
> .
>